### PR TITLE
fix: create-position

### PR DESCRIPTION
**Pull request by Senna46**
Fix create position problem

### CODE REVIEW

**Comment by Senna46** [comment]
@taiki1frsh 
Fixed the problem of negative margins when creating positions.
Changed that Levy Period does not occur for 8 hours after opening a position.

Plz review this PR & #498

File: x/derivatives/types/positions.go [hunk 1]
```diff
@@ -234,11 +234,14 @@ func (m PerpetualFuturesPosition) RequiredMarginInBase() sdk.Int {
 func (m PerpetualFuturesPosition) RequiredMarginInMetrics(baseMetricsRate, quoteMetricsRate MetricsRateType) sdk.Int {
 	// 必要証拠金(USD単位) = 必要証拠金(quote単位) * 現在のquote/USDレート
 	//                    = 必要証拠金(base単位) * 現在のbase/USDレート
+	// 必要証拠金(uusd単位) = 必要証拠金(USD単位) * 1000000
 	if m.RemainingMargin.Denom == m.Market.QuoteDenom {
 		baseQuoteRate := baseMetricsRate.Amount.Amount.Quo(quoteMetricsRate.Amount.Amount)
-		return sdk.NewDecFromInt(m.RequiredMarginInQuote(baseQuoteRate)).Mul(quoteMetricsRate.Amount.Amount).TruncateInt()
+		usdAmount := sdk.NewDecFromInt(m.RequiredMarginInQuote(baseQuoteRate)).Mul(quoteMetricsRate.Amount.Amount)
+		return NormalToMicroInt(usdAmount)
```
Comment (taiki1frsh) [inline x/derivatives/types/positions.go:241]: In `MarginRequirement`, make the unit micro is already done.
Is this required?

**Comment by taiki1frsh** [comment]
@Senna46 
Please write unit tests for `RequiredMarginInMetrics`.

Also, try running unit tests for the function you modify.

You can realize that `TestPerpetualFuturesPosition_CalcProfitAndLoss` is not working with this modification.
`TestCalcReturningAmountAtClose` as well.

File: x/derivatives/types/positions.go [hunk 2]
```diff
@@ -234,11 +234,14 @@ func (m PerpetualFuturesPosition) RequiredMarginInBase() sdk.Int {
 func (m PerpetualFuturesPosition) RequiredMarginInMetrics(baseMetricsRate, quoteMetricsRate MetricsRateType) sdk.Int {
```
Comment (taiki1frsh) [inline x/derivatives/types/positions.go:234]: I think InMetrics means in "usd" as the comment says.

If you modify it in "uusd" (micro metrics), change the function name.

File: x/derivatives/types/positions.go [hunk 3]
```diff
@@ -302,28 +305,34 @@ func (m PerpetualFuturesPosition) ProfitAndLossInQuote(baseMetricsRate, quoteMet
 
 func (m PerpetualFuturesPosition) ProfitAndLossInMetrics(baseMetricsRate, quoteMetricsRate MetricsRateType) sdk.Int {
```
Comment (taiki1frsh) [inline x/derivatives/types/positions.go:306]: I think InMetrics means in "usd" as the comment says.

If you modify it in "uusd" (micro metrics), change the function name.

File: x/derivatives/types/positions.go [hunk 4]
```diff
@@ -302,28 +305,34 @@ func (m PerpetualFuturesPosition) ProfitAndLossInQuote(baseMetricsRate, quoteMet
...
 
 func (m PerpetualFuturesPosition) RemainingMarginInQuote(quoteMetricsRate MetricsRateType) sdk.Int {
 	// 残存証拠金(USD単位) = 残存証拠金(quote単位) * 現在のquote/USDレート
-	return sdk.NewDecFromInt(m.RemainingMargin.Amount).Mul(quoteMetricsRate.Amount.Amount).TruncateInt()
+	result := sdk.NewDecFromInt(m.RemainingMargin.Amount).Mul(quoteMetricsRate.Amount.Amount)
+	// uusd単位に変換
+	return NormalToMicroInt(result)
 }
 
 func (m PerpetualFuturesPosition) RemainingMarginInMetrics(baseMetricsRate, quoteMetricsRate MetricsRateType) sdk.Int {
```
Comment (taiki1frsh) [inline x/derivatives/types/positions.go:333]: I think InMetrics means in "usd" as the comment says.

If you modify it in "uusd" (micro metrics), change the function name.

**Pull request closed without merging**